### PR TITLE
Adding `release.yml` for CI release job

### PR DESCRIPTION
**Pull request by ktsitsi**
This PR:

- Introduces a `release.yml` for releasing the project on PyPI registry, using a trusted publisher.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Release
+on:
+  release:
+    types: [ 'published' ]
+
```
Comment (johnkerl) [inline .github/workflows/release.yml:5]: as a thought -- in tiledbsoma we have a publish to test-pypi for things like manual runs, nightlies, etc -- enabling us to catch issues _before_ taggint the release

Comment (ktsitsi) [reply .github/workflows/release.yml:5]: Sure I can add that.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  pypi-publish:
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    environment: pypi
+    permissions:
+      id-token: write
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
```
Comment (bkmartinjr) [inline .github/workflows/release.yml:15]: suggest using latest versions of the actions - this one is quite old

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (bkmartinjr) [inline .github/workflows/release.yml:17]: use latest (v4)

**Review by bkmartinjr — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  pypi-publish:
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    environment: pypi
+    permissions:
+      id-token: write
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v4
```
Comment (ryan-williams) [inline .github/workflows/release.yml:15]:
```suggestion
        uses: actions/setup-python@v5
```
(optional)

**Review by ryan-williams — APPROVED** [review]

**Pull request merged**